src/Model/MyParcelCustomsItem.php

```diff
@@ -48,14 +53,25 @@ public function getDescription()
      * Required: Yes
      *
      * @param mixed $description
+     * @param string|null $carrierName
      * @return $this
+     *
+     * @throws \Exception
      */
-    public function setDescription($description)
+    public function setDescription($description, ?string $carrierName = null): self
     {
-        /**
-         * Description cut after 47 chars
-         */
-        $this->description = Str::limit($description, self::DESCRIPTION_MAX_LENGTH);
+        $maxLength = AbstractConsignment::DESCRIPTION_MAX_LENGTH;
+
+        if ($carrierName) {
+            $carrier   = ConsignmentFactory::createByCarrierName($carrierName);
```

Comment:
Als je de `$carrierName` param verandert in `$carrier` (`string|int|AbstractCarrier`) en dan `ConsignmentFactory::createFromCarrier($carrier)` gebruikt maak je dit een stuk prettiger voor gebruikers, dan kan je namelijk een carrier id, naam of class(/class-string) meegeven.